File: backend/Domain/Entities/BookingHeader.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entities
{
    [Table("adm_BookingHeader")]
    public class BookingHeader : BaseAuditable
    {
        [Required, ForeignKey(nameof(AmenityMaster))]
        public long AmenityId { get; set; }

        [Required, ForeignKey(nameof(Society))]
        public long SocietyId { get; set; }

        [Required]
        [MaxLength(50)]
        public string BookingNo { get; set; } = default!;

        [Required]
        public DateTime BookingDate { get; set; }

        [MaxLength(500)]
        public string? Remarks { get; set; }

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "Draft";

        public long? ResidentUserId { get; set; }

        public long? FlatId { get; set; }

        [MaxLength(200)]
        public string? ResidentNameSnapshot { get; set; }

        [MaxLength(30)]
        public string? ContactNumberSnapshot { get; set; }

        public bool? IsChargeableSnapshot { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? AmountBeforeTax { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? TaxAmount { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? DepositAmount { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? DiscountAmount { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? ConvenienceFee { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? TotalPayable { get; set; }

        public bool? RequiresApprovalSnapshot { get; set; }

        public long? ApprovedBy { get; set; }

        public DateTime? ApprovedOn { get; set; }

        [MaxLength(500)]
        public string? RejectionReason { get; set; }

        public long? CancelledBy { get; set; }

        public DateTime? CancelledOn { get; set; }

        [MaxLength(500)]
        public string? CancellationReason { get; set; }

        [MaxLength(50)]
        public string? RefundStatus { get; set; }

        public AmenityMaster AmenityMaster { get; set; } = default!;
        public Property Society { get; set; } = default!;
        public ICollection<BookingUnit> BookingUnits { get; set; } = new List<BookingUnit>();
    }
}

File: frontend/src/app/model/booking-header.ts
export interface BookingUnit {
    id?: number;
    bookingId?: number;
    amenityUnitId?: number;
    unitNameSnapshot?: string;
    capacityReserved?: number;
    bookingSlot?: BookingSlot | null;
}

export interface BookingSlot {
    id?: number;
    bookingId?: number;
    bookingUnitId?: number;
    amenityId?: number;
    amenityUnitId?: number;
    slotStartDateTime?: string;
    slotEndDateTime?: string;
    slotStatus?: string;
    checkInRequired?: boolean;
    checkInTime?: string;
    checkOutTime?: string;
}

export interface BookingSlotAvailability {
    id?: number;
    slotStartDateTime: string;
    slotEndDateTime: string;
    capacityPerSlot: number;
    availableCapacity: number;
    slotCharge?: number;
    isChargeable: boolean;
    chargeType?: string;
    baseRate?: number;
    securityDeposit?: number;
    refundableDeposit: boolean;
    taxApplicable: boolean;
    taxCodeId?: number;
    taxPercentage?: number;
}

export class BookingHeader {
    id?: number;
    amenityId?: number;
    amenityName?: string;
    societyId?: number;
    societyName?: string;
    bookingNo?: string;
    bookingDate?: string;
    remarks?: string;
    status?: string;
    residentUserId?: number;
    flatId?: number;
    residentNameSnapshot?: string;
    contactNumberSnapshot?: string;
    isChargeableSnapshot?: boolean;
    amountBeforeTax?: number;
    taxAmount?: number;
    depositAmount?: number;
    discountAmount?: number;
    convenienceFee?: number;
    totalPayable?: number;
    requiresApprovalSnapshot?: boolean;
    approvedBy?: number;
    approvedOn?: string;
    rejectionReason?: string;
    cancelledBy?: number;
    cancelledOn?: string;
    cancellationReason?: string;
    refundStatus?: string;
    createdDate?: string;
    createdBy?: number;
    modifiedDate?: string;
    modifiedBy?: number;
    bookingUnits?: BookingUnit[];
}
